cbor.js

```diff
@@ -50,12 +104,43 @@ function encode(value) {
     lastLength = length;
     return dataView;
   }
+
   function commitWrite() {
     offset += lastLength;
   }
-  function writeFloat64(value) {
-    commitWrite(prepareWrite(8).setFloat64(offset, value));
+
+  function encodeFloat16(value) {
+    var dataView = prepareWrite(3);
+    dataView.setUint8(offset, 0xf9);
+    dataView.setUint16(offset+1, getFloat16(value));
+    commitWrite(3);
+  }
+  
+  function encodeFloat(value) {
+    var dataView;
+    var f16 = checkFloat16(value);
+    if (f16 !== false) {
+      dataView = prepareWrite(3);
+      dataView.setUint8(offset, 0xf9);
+      dataView.setUint16(offset+1, f16);
+      commitWrite();
+      return;
+    }
+    var f32 = checkFloat32(value);
+    if (f32 !== false) {
+      dataView = prepareWrite(5);
+      dataView.setUint8(offset, 0xfa);
+      dataView.setUint32(offset+1, f32);
```

Comment:
I'd like to use the native encoding functions wherever possible. I'd prefer to use `dataView.setFloat32` instead of encoding the float to uint32 ourself.  That's another reason for changing the "fits into float16/float32"-check-function.